From cc37164f7d56f5593de5f85150099afa9af19adf Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Mon, 2 Jan 2017 14:36:29 +1100
Subject: [PATCH 32/33] dpnet: Map response Data

This allows Civ III to see a server on the next working.

Needs to be merged into the EnumHost onces.
---
 dlls/dpnet/client.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/dlls/dpnet/client.c b/dlls/dpnet/client.c
index 908b240ed3..6e3a71038f 100644
--- a/dlls/dpnet/client.c
+++ b/dlls/dpnet/client.c
@@ -570,7 +570,7 @@ void process_enumhost(SOCKET sock, struct sockaddr_in host, IDirectPlay8Address
                     {
                         struct ENUM_QUERY_RESPONSE *data = (struct ENUM_QUERY_RESPONSE *)header;
 
-                        TRACE("PACKET_ENUM_RESPONSE (%d). (%d)\n", sizeof(struct ENUM_QUERY_RESPONSE) received);
+                        TRACE("PACKET_ENUM_RESPONSE (%lu), (%d)\n", sizeof(struct ENUM_QUERY_RESPONSE), received);
 
                         if(received <= sizeof(struct ENUM_QUERY_RESPONSE))
                         {
@@ -605,6 +605,13 @@ void process_enumhost(SOCKET sock, struct sockaddr_in host, IDirectPlay8Address
                             response.pApplicationDescription = &desc;
                             response.pAddressDevice          = device;
 
+                            if(data->reply_offset < received)
+                            {
+                                response.pvResponseData = ((char*)data)+data->reply_offset+sizeof(struct ENUM_HEADER);
+                                response.dwResponseDataSize = data->response_size;
+                            }
+
+
                             desc.guidInstance     = data->instance;
                             desc.guidApplication  = data->application;
                             desc.dwCurrentPlayers = data->current_players;
-- 
2.11.0

